In [1]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier, AdaBoostClassifier
from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, f1_score
from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
from xgboost import XGBClassifier
from sklearn import svm, datasets

np.random.seed(1)

C:\Users\Kanna\anaconda3\lib\site-packages\xgboost\compat.py:36: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
  from pandas import MultiIndex, Int64Index


In [2]:
df = pd.read_csv('C:/Users/Kanna/Downloads/heartfailure_train_df.csv')
df.head(5)

AGE   ANAEMIA  CREATININE_PHOSPHOKINASE  DIABETES  EJECTION_FRACTION  \
0 -1.001659  1.084935                 -0.475559 -0.827940          -0.687067   
1  0.734629 -0.921714                 -0.353834  1.207818          -1.765496   
2  1.148031 -0.921714                 -0.439595 -0.827940           0.972055   
3 -0.092175  1.084935                 -0.007104  1.207818           1.801616   
4 -0.918978  1.084935                  1.602981  1.207818          -0.272287   

   HIGH_BLOOD_PRESSURE  PLATELETS  SERUM_CREATININE  SERUM_SODIUM       SEX  \
0             1.322876   1.751802         -0.400321      0.333046 -1.350676   
1             1.322876   1.352637         -0.400321     -0.118945  0.740370   
2             1.322876  -0.128478         -0.312712      2.593001  0.740370   
3            -0.755929  -0.695713         -0.312712      1.237028 -1.350676   
4            -0.755929  -1.945732         -0.487931      1.237028 -1.350676   

    SMOKING      TIME  DEATH_EVENT  
0 -0.664364 -1.514594    -0.709646  
1  1.505199  0.754398    -0.709646  
2 -0.664364  1.025130    -0.709646  
3 -0.664364 -1.243862     1.409154  
4 -0.664364 -0.044906     1.409154

In [3]:
train_X = pd.read_csv('C:/Users/Kanna/Downloads/heartfailure_train_X.csv')
train_y =pd.read_csv('C:/Users/Kanna/Downloads/heartfailure_train_y.csv') 
test_X = pd.read_csv('C:/Users/Kanna/Downloads/heartfailure_test_X.csv')
test_y = pd.read_csv('C:/Users/Kanna/Downloads/heartfailure_test_y.csv')

In [4]:
test_y

DEATH_EVENT
0              0
1              0
2              0
3              0
4              0
..           ...
294            1
295            1
296            1
297            1
298            1

[299 rows x 1 columns]

# logistic regression default


In [5]:
model = LogisticRegression()

In [6]:
lgr = model.fit(train_X, train_y)

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [20]:
y_pred = model.predict(test_X)
accuracy = accuracy_score(test_y, y_pred)
recall=recall_score(test_y, y_pred)
print("recall:", recall)
print("Accuracy:", accuracy)

recall: 0.9211822660098522
Accuracy: 0.8528428093645485


# logistic regression using random and grid

In [15]:
lgr = LogisticRegression()

# Define the hyperparameters that you want to search over using a grid search
grid_param = {'penalty': ['l1', 'l2'], 
              'C': [0.001, 0.01, 0.1, 1, 10, 100],
              'solver': ['liblinear', 'saga']}

# Define the hyperparameters that you want to search over using a random search
random_param = {'penalty': ['l1', 'l2'], 
                'C': np.logspace(-4, 4, 20),
                'solver': ['liblinear', 'saga']}

# Perform a grid search to find the best hyperparameters
grid_search = GridSearchCV(lgr, grid_param, cv=5)
grid_search.fit(train_X, train_y)
best_grid = grid_search.best_estimator_

# Perform a random search to find the best hyperparameters
random_search = RandomizedSearchCV(lgr, random_param, cv=5)
random_search.fit(train_X, train_y)
best_random = random_search.best_estimator_

# Evaluate the models on the test set
y_pred_grid = best_grid.predict(test_X)
y_pred_random = best_random.predict(test_X)

acc_grid = accuracy_score(test_y, y_pred_grid)
acc_random = accuracy_score(test_y, y_pred_random)

print("Accuracy using grid search:", acc_grid)
print("Accuracy using random search:", acc_random)

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

Accuracy using grid search: 0.8394648829431438
Accuracy using random search: 0.8494983277591973


C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

# decision tree default

In [22]:
dtclf = DecisionTreeClassifier(random_state=42)

# train the classifier on the training data
dtclf.fit(train_X, train_y)

# make predictions on the testing data
y_pred = dtclf.predict(test_X)

# calculate accuracy of the classifier
accuracy = accuracy_score(test_y, y_pred)


print("Accuracy: {:.2f}%".format(accuracy*100))

Accuracy: 100.00%


# decision tree using random search and grid search

In [24]:
param_grid = {
    'criterion': ['gini', 'entropy'],
    'max_depth': [2, 5, 10, None],
    'min_samples_split': [2, 5, 10],
    'min_samples_leaf': [1, 2, 4],
}

# Define random search parameters
random_param = {
    'criterion': ['gini', 'entropy'],
    'max_depth': np.arange(2, 20),
    'min_samples_split': np.arange(2, 20),
    'min_samples_leaf': np.arange(1, 10),
}

# Create decision tree classifier
dtc = DecisionTreeClassifier()

# Perform grid search
grid_search = GridSearchCV(estimator=dtc, param_grid=param_grid, cv=5)
grid_search.fit(train_X, train_y)
print('Best parameters for grid search:', grid_search.best_params_)
print("Accuracy: {:.2f}%".format(accuracy*100))

# Perform random search
random_search = RandomizedSearchCV(estimator=dtc, param_distributions=random_param, n_iter=50, cv=5)
random_search.fit(train_X, train_y)
print('Best parameters for random search:', random_search.best_params_)
y_pred = random_search.predict(test_X)

accuracy = accuracy_score(test_y, y_pred)
print("Accuracy: {:.2f}%".format(accuracy*100))


Best parameters for grid search: {'criterion': 'gini', 'max_depth': 2, 'min_samples_leaf': 1, 'min_samples_split': 2}
Accuracy: 84.62%
Best parameters for random search: {'min_samples_split': 15, 'min_samples_leaf': 3, 'max_depth': 2, 'criterion': 'gini'}
Accuracy: 84.62%


# svm using random search and grid search


In [14]:
# Define the parameter grid for Grid Search
param_grid = {'C': [0.1, 1, 10, 100], 'gamma': [0.001, 0.01, 0.1, 1], 'kernel': ['linear', 'rbf', 'poly']}

# Define the parameter distributions for Random Search
param_dist = {'C': np.logspace(-3, 3, 7), 'gamma': np.logspace(-3, 3, 7), 'kernel': ['linear', 'rbf', 'poly']}

# Initialize SVM classifier
svm_classifier = svm.SVC()

# Perform Grid Search
grid_search = GridSearchCV(svm_classifier, param_grid=param_grid, cv=5)
grid_search.fit(train_X, train_y)

# Perform Random Search
random_search = RandomizedSearchCV(svm_classifier, param_distributions=param_dist, n_iter=20, cv=5)
random_search.fit(train_X, train_y)

# Print the best hyperparameters found by Grid Search
print("Best hyperparameters using Grid Search: ", grid_search.best_params_)

# Print the best hyperparameters found by Random Search
print("Best hyperparameters using Random Search: ", random_search.best_params_)

# Evaluate the model using the best hyperparameters found by Grid Search
svm_classifier.set_params(**grid_search.best_params_)
svm_classifier.fit(train_X, train_y)
print("Accuracy using best hyperparameters found by Grid Search: ", svm_classifier.score(test_X, test_y))

# Evaluate the model using the best hyperparameters found by Random Search
svm_classifier.set_params(**random_search.best_params_)
svm_classifier.fit(train_X, train_y)
print("Accuracy using best hyperparameters found by Random Search: ", svm_classifier.score(test_X, test_y))

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), 

C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


Best hyperparameters using Grid Search:  {'C': 1, 'gamma': 0.001, 'kernel': 'linear'}
Best hyperparameters using Random Search:  {'kernel': 'linear', 'gamma': 0.01, 'C': 100.0}
Accuracy using best hyperparameters found by Grid Search:  0.8461538461538461


C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Users\Kanna\anaconda3\lib\site-packages\sklearn\utils\validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


Accuracy using best hyperparameters found by Random Search:  0.8494983277591973


# Summary

After training and splitting the data in the previous notebook we have directly imported and loaded the data sets of train and test, now that we have loaded the data we can directly perfomr the metric analysis. For this I have taken accuracy as my metrics because as per my data set is considered it is well balanced and trained and after the changes made to 0 and 1 which says that in the target varibale, if it says the value is 0 then the death is not occured and if it says 1 death is occured.

1)To predict the death occured precisely considering the other attributes I have choosen accuracy as my scoring metric. As a classification problem I have first made inclinatio to do the logistic regression using random search and grid search for which I have closely got to 84% in both the cases, based on the problem that we are solving I am satisfied that the data is balanced and have acheived good accuracy for logistic regression.


2)Decision tree: For decision tree default we have achieved an accuracy score of 100 percent which is good thing to consider, we understand the training data has leant to overfit the data and also it has leanrt the noise and peculiarities of the training data set. But one thought that strikes here is that the training data set has learnt good enough that is giving us better output but we should also use cross validation techniques than can genralise well on the new data obtained from the training data.

3)IN SVM using random search and grid search the model is able to give the accuracy if 84 percent in both the cases that means the depth of the data that this model is covering to get 84 percent is a good, as explained above the training data set have learned to overcome the noise and and in this case it can be considered as a good model for the quality of the data that we have after training the data.

Overall, the accuracy acheived is nearly 84 percent in all the cases and we can conclude that the data is trained well and the quality of the data is good, but what I still feel is that we can tune the hyper parameters and can acheive good results above 84 percent, there is also decent amount of chance that the model may not be able to generalize to new data that is significantly different from the training data. To evaluate all this we should tune the hyperparametrs and also check thorougly with the overfitting of the data.a